### ISSUE TITLE
Add project - BosnD

### ISSUE THREAD

**Issue by m4r10k** [veggiemonk/awesome-docker#642]
Hi,

I would like to add a link. https://gitlab.com/n0r1sk/bosnd

*BosnD, the boatswain daemon. Dynamic, Docker Swarm / Kubernetes / File based configuration file templating for all kinds of services (Apache httpd, Nginx, HAProxy, ...) made real.*

We @rauschbit, @cooperspencer and I are the authors of this project and we use it heavily at [STRABAG SE](https://www.strabag.com/) to provide automatic load balancer configuration based on the Docker Swarm service discovery information for more than 300 load balancers (Apache httpd, Nginx, ...)

The readme of the project is complete for the recent version and almost complete for the upcoming release.

Unfortunately I am not sure where it would be fitted in best, maybe `Deployment and Infrastructure` as it could be used for any kind of file configurable binary? We are open for any suggestions on this.

The project is hosted at GitLab, as we are using their integrated build pipelines for other projects.

Thanks for looking at this and thank you for maintaining this list :hugs: ! 

Mario

**Comment by veggiemonk** [comment]
Hi @m4r10k 

Sure, I would love to merge this.
The problem is that there is no instructions on how to install or use the software. 
I read the readme and it seems like there is a lot of details but how can I test that without reading the code. We value people's time so we want to make it really easy for people to test and try the project.

Sorry for the delay, I was at KubeCon.

You can open a PR or ping me here when the instructions have been updated.

Keep up the good work!

**Comment by m4r10k** [comment]
@veggiemonk I added a Quickstart tutorial which shows the most simplest usage by providing enough information and ideas about whats going on. A short Lessons Learned is also there now. It should be easy to follow. Let me know, if you have any further suggestions for the first admission to the list! :smiley:

**Comment by veggiemonk** [comment]
Hi @m4r10k, 
nice job! 
I can see you've put a good amount of time on that and it looks great.

Although, I still don't understand what it does or *why* would I use that project, meaning: **What problem does it solves** 

If I understand correctly, it is a templating engine that reconfigure services ...... But even that I'm not sure.
I think it would be good if you can take a step back and imagine explaining this to someone who doesn't know docker or swarm! 
I totally understand when working on something that we get too close the code. 
The fact that I don't know in which category this would belong makes it hard to add it to the list 😄 

Keep up the good work, you're almost there.

Also, can you please create a PR for it?

Thank you for your patience 🙏

**Comment by rauschbit** [comment]
Hmm, I think the "What it does" section describes the use case really good - what's unclear on that for you? Would be helpful to write a better "What it does" description.  :)

The BosnD is a service which gathers information from a Docker Swarm environment (by itself or via the midshipman) about the running services/ip's of containers in a stack/stack's overlay network to take this information in combination with configuration files (from services like apache httpd/nginx/...) where the corresponding parts (like for nginx the "upstream" backends, or for apache httpd the "proxypass" backends) are configured via go templating language. for example: the configuration files of the services are enhanced by dynamic go-template parts to dynamically configure the backends for a reverse proxy.
The outcome is a dynamically written configuration file for services like apache httpd/nginx/... + an automatic reload of this services when something in your dynamic docker swarm stack changes (like the services tasks [containers] scale up or down).

I hope that describes it better - we will add that to our description if that's now clearer for you.
To describe the service for someone who doesn't know what docker or swarm is, that would be really hard imo. Maybe "dynamic configuration file writer & service reloader for dynamically changing container environments" would be the headline.

Because one can use it for many purposes, we, for now, use it for reverse proxy configurations for our apache httpds & nginx reverse proxies - but the BosnD isn't limited to those services...

We will make the PR if you say that you now understand the purpose of our service.  :)

Thx for your input!

**Comment by veggiemonk** [comment]
What is unclear is that the "what it does" describes "how it does it".

Look at this repo: https://github.com/jwilder/nginx-proxy

```
nginx-proxy sets up a container running nginx and docker-gen. docker-gen generates reverse proxy configs for nginx and reloads nginx when containers are started and stopped.
```

For some reason, I can understand that description ☝️ .

```
BosnD, the boatswain daemon. Dynamic, Docker Swarm / Kubernetes / File based configuration file templating for all kinds of services (Apache httpd, Nginx, HAProxy, ...) made real.
```

I don't understand because it is trying to explain too many things. What does `made real` means?

So the description is better:  "dynamic configuration file writer & service reloader for dynamically changing container environments"

If BosnD is a reverse-proxy, it can be added to the reverse proxy section. 

Let me know if you need help to make the PR 😄

**Comment by m4r10k** [comment]
@veggiemonk Thank you for your feedback! I updated the description as it is really better this way! 😄

**Comment by veggiemonk** [comment]
Merged by #648
